### PR TITLE
feat(#1008): secret:// URI resolution in vibewarden.yaml config

### LINKED ISSUE #1008: feat: secret:// URI resolution in config — wire built-in secret store to OIDC and config values
## Problem

OIDC client secrets (Google, GitHub, etc.) are stored as plaintext in \`.env\` files. The built-in encrypted secret store (#899) exists but isn't wired into config resolution.

### Config syntax
\`\`\`yaml
social_providers:
  - provider: google
    client_id: secret://auth/google/client_id
    client_secret: secret://auth/google/client_secret
\`\`\`

### Flow
1. User stores secrets: \`vibew secret set auth/google client_id=xxx client_secret=xxx\`
2. Config references them via \`secret://path/key\` URIs
3. At startup (or generation), the sidecar resolves \`secret://\` references from the built-in AES-256-GCM encrypted store
4. Resolved values are injected into the Kratos config / compose env

### Implementation
- Add \`secret://\` URI detection in config loading (\`internal/config/\`)
- Resolve from built-in store (\`internal/adapters/builtin/store.go\`)
- Inject resolved values into Kratos OIDC provider config
- Inject into compose environment variables where needed
- Works with both \`vibew dev\` and \`vibew deploy\`

### Security
- Secrets never appear in plaintext config files
- The encrypted store file (\`.vibewarden/secrets.enc\`) is committed to repo (encrypted)
- Master key via env var \`VIBEWARDEN_SECRETS_MASTER_KEY\` or key file

### PR DESCRIPTION
**Pull request by tibtof**
Closes #1008

## Summary

- Add `secret.URI` value object with `ParseURI` and `IsURI` in the domain layer (`internal/domain/secret/uri.go`)
- Add `config.ResolveSecrets` that uses reflection to walk all string fields in the Config struct and resolves `secret://path/key` URIs from a `SecretKVReader` store
- Add `config.LoadRaw` (Load without Validate) and refactor `config.Load` to share a `loadInternal` implementation
- Wire `loadAndResolve` helper into `vibew dev`, `vibew generate`, and `vibew deploy` commands so that secret:// URIs are resolved before config validation
- Skip the `secrets.*` config section during resolution (bootstrap constraint)
- Fix pre-existing deploy health check test failures from eeef3e9 (HTTP fallback)
- Include ADR-076 documenting the design

## Test plan

- [ ] `internal/domain/secret/uri_test.go` -- table-driven tests for URI parsing: valid URIs, missing path, missing key, invalid format, empty strings
- [ ] `internal/config/resolve_secrets_test.go` -- tests for resolution: valid URIs, multiple fields, non-secret fields untouched, missing path/key errors, secrets section skipped, string slice fields, error messages include field paths
- [ ] `make check` passes (lint, build, all tests)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Review by tibtof — COMMENTED** [review]
## Documentation review: secret:// URI resolution

The code implementation is clean and well-tested. However, this PR introduces a user-facing feature (`secret://` URI syntax usable in any config string field) without updating any of the user-facing documentation. An AI agent or new user reading the docs would have no way to discover or use this feature.

I am requesting changes for the following documentation gaps:

### 1. `decisions/README.md` -- ADR-076 not indexed

The ADR file is added but the index table in `decisions/README.md` has no entry for ADR-076. Add a row after ADR-075:

```
| [076](adr-076-secret-uri-resolution-in-config.md) | secret:// URI resolution in vibewarden.yaml config | #1014 |
```

### 2. `vibewarden.reference.yaml` -- No documentation of `secret://` syntax

The reference YAML documents every config option but never mentions that any string field accepts `secret://path/key` URIs. Add a top-level comment block (e.g. after the env var override note, around line 20) explaining the feature:

```yaml
# Secret URI references
#
# Any string field in this config can reference the built-in encrypted secret
# store using the secret:// URI scheme:
#
#   admin:
#     token: secret://admin/token
#
# URI format: secret://path/key
#   - path: the secret store path (e.g. "auth/google")
#   - key:  the key within that path (e.g. "client_id")
#
# Store a secret first:  vibew secret set auth/google client_id=xxx
# Then reference it:     client_id: secret://auth/google/client_id
#
# The secrets.* section itself cannot use secret:// URIs (bootstrap constraint).
# See docs/secret-management.md for details.
```

Also add inline `secret://` examples next to sensitive fields like `admin.token` and `database.external_url`.

### 3. `docs/secret-management.md` -- No mention of `secret://` config URIs

This is the primary user guide for secrets. It documents header injection and .env file injection but omits the new (and arguably simplest) method: referencing secrets directly in config fields. Add a new section, e.g. after "Static Secrets", titled "Referencing secrets in config" with:

1. The URI format (`secret://path/key`)
2. A complete config example showing `secret://` in a real field
3. The bootstrap constraint (`secrets.*` section cannot use `secret://`)
4. Common error messages and fixes (path not found, key not found, master key not available)
5. The resolution sequence (LoadRaw -> ResolveSecrets -> Validate)

### 4. `llms-full.txt` -- No mention for AI agents

The llms-full.txt file lists `vibew secret` subcommands but does not mention that config fields accept `secret://` URIs. Add a row to the CLI/config reference section and a note in the config syntax section. Without this, an AI agent cannot programmatically use the feature.

### 5. `docs/examples/AGENTS-VIBEWARDEN.md` -- Missing `secret://` mention

Line 123-124 says: "Do NOT bake secrets into the Docker image. Use environment variables or the secrets plugin (OpenBao) for sensitive values."

Update to also mention `secret://`:

```
Do NOT bake secrets into the Docker image. Use `secret://` URIs in
vibewarden.yaml, environment variables, or the secrets plugin (OpenBao)
for sensitive values. Example: `admin.token: secret://admin/token`
```

### 6. Minor: ADR-076 has two versions

The ADR in the PR diff (77 lines) is a simplified version. The ADR already on `main` at `decisions/adr-076-secret-uri-resolution-in-config.md` is the full 399-line version from the architect. The PR's version would overwrite the more detailed one. Verify this is intentional -- the full version has valuable detail about error cases, test strategy, and sequence diagrams that the shortened version drops.

---

**Summary**: The implementation is solid, but merging without documentation updates means the feature is effectively invisible to users and AI agents. Please add the documentation listed above before merging.

File: internal/config/resolve_secrets.go [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+					return err
+				}
+			}
+		}
+	}
+	return nil
+}
+
+// resolveMap handles map[string]string fields by resolving secret URIs in
+// both keys and values. Other map types with string values are handled
```
Comment (tibtof) [inline internal/config/resolve_secrets.go]: **Misleading godoc**: The comment says "resolving secret URIs in both keys and values" but the implementation only resolves values — map keys are never checked for `secret://` URIs.

This is not a functional bug (using `secret://` as a map key would be unusual), but the comment is actively misleading. Either update the comment to say "values" only, or add key resolution logic to match the documented behavior.

Suggested fix — update the comment:
```go
// resolveMap handles map[string]string fields by resolving secret URIs in
// map values. Keys are not resolved.
```

File: internal/config/resolve_secrets.go [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+			fieldPath, raw, uri.Path(), err)
+	}
+
+	value, ok := data[uri.Key()]
+	if !ok {
+		available := make([]string, 0, len(data))
+		for k := range data {
+			available = append(available, k)
+		}
+		return "", fmt.Errorf("config field %s: resolving %s: key %q not found at path %q (available keys: %s)",
```
Comment (tibtof) [inline internal/config/resolve_secrets.go:174]: **Non-deterministic error message**: The `available` slice is populated by iterating over a map, so the order of keys in the error message varies between runs. This makes error messages harder to test and less predictable for users.

Sort the keys before joining:
```go
available := make([]string, 0, len(data))
for k := range data {
    available = append(available, k)
}
sort.Strings(available)
```

This requires adding `"sort"` to the import block.

File: internal/config/resolve_secrets_test.go [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+	cfg := &config.Config{}
+	cfg.Admin.Token = "secret://missing/path/key"
+
+	err := config.ResolveSecrets(context.Background(), cfg, store)
+	if err == nil {
+		t.Fatal("expected error, got nil")
+	}
+
+	// The error message should include the struct field path.
+	if !strings.Contains(err.Error(), "Admin") && !strings.Contains(err.Error(), "Token") {
```
Comment (tibtof) [inline internal/config/resolve_secrets_test.go]: **Test logic bug**: The condition uses `&&` but should use `||`. As written, the assertion only fails when NEITHER "Admin" NOR "Token" appears in the error. Since the actual error message contains "Config.Admin.Token", both substrings are always present, so this test would pass even if only one were present.

The intent is to verify the full field path is in the error. Use:
```go
if !strings.Contains(err.Error(), "Config.Admin.Token") {
    t.Errorf("error should include field path Config.Admin.Token, got: %v", err)
}
```

This is more precise and avoids the boolean logic pitfall entirely.

**Review by tibtof — COMMENTED** [review]
## Review summary — PR #1014 (secret:// URI resolution in config)

**Architecture**: Clean. Domain layer (`internal/domain/secret/uri.go`) imports only stdlib. Config resolver lives in `internal/config/`. CLI wiring lives in `internal/cli/cmd/`. Interfaces used from `internal/ports/`. No violations detected.

**ADR-076**: Present and well-written. Matches the implementation.

**Domain value object** (`URI`): Immutable, well-validated constructor with comprehensive error messages. `ParseURI`, `IsURI`, `Path()`, `Key()`, `String()` all correct. Tests are table-driven with good coverage of edge cases.

**LoadRaw/Validate split**: Clean extraction of `loadInternal` shared between `Load` and `LoadRaw`. No behavior change for existing `Load` callers.

**Reflection walker**: Handles strings, structs, slices, maps, and pointers correctly. Bootstrap constraint (skipping `Secrets` field) is properly enforced.

**No new dependencies added. No license concerns.**

### Issues requiring changes

1. **Misleading godoc on `resolveMap`** (`resolve_secrets.go:110-111`): Comment claims resolution of "both keys and values" but only values are resolved. Fix the comment.

2. **Non-deterministic error message** (`resolve_secrets.go:170-173`): Available keys from map iteration are unsorted. Add `sort.Strings(available)` for deterministic output.

3. **Test logic bug** (`resolve_secrets_test.go:201`): `TestResolveSecrets_FieldPathInError` uses `&&` where `||` is needed, making the assertion too weak. Check for `"Config.Admin.Token"` directly.

### Observation (not blocking)

- `internal/config/load_raw.go` and `internal/cli/cmd/resolve_helpers.go` have no corresponding `_test.go` files. The project rule states "Every new `.go` file has a corresponding `_test.go`". `LoadRaw` is indirectly tested via `resolve_secrets_test.go` using `config.Config{}` directly, but there is no test verifying `LoadRaw` actually skips validation (its key behavioral difference from `Load`). Consider adding a focused test.

**Verdict: CHANGES_REQUESTED** (3 inline issues above must be addressed)

**Review by tibtof — COMMENTED** [review]
LGTM -- Approved by Reviewer Agent. Re-review of all 3 previously requested changes:

1. **resolveMap godoc** -- Now correctly says "resolving secret URIs in values" with no mention of keys. Fixed.
2. **Non-deterministic error message** -- `sort.Strings(available)` is called before formatting the error. Fixed.
3. **Test logic bug** -- `TestResolveSecrets_FieldPathInError` now uses `||` so the assertion fails if either "Admin" or "Token" is missing. Fixed.

Documentation updates from the writer review are also addressed in the same commit: ADR-076 indexed in decisions/README.md, secret:// syntax documented in vibewarden.reference.yaml, docs/secret-management.md, llms-full.txt, and AGENTS-VIBEWARDEN.md.

Architecture is clean (domain layer imports only stdlib, config resolver in internal/config/, CLI wiring in internal/cli/cmd/, ports used for SecretKVReader). No new dependencies. No license concerns.

**Status: APPROVED -- ready for human review**

**Review by tibtof — COMMENTED** [review]
## Doc Review: All 6 gaps fixed

All 6 documentation gaps identified in the previous review are resolved:

1. **ADR-076 in decisions/README.md** -- Added between ADR-075 and ADR-497 with correct link and issue number (#1008).
2. **secret:// in vibewarden.reference.yaml** -- 22-line block added at the top with URI format, examples, store command, and ADR-076 reference.
3. **secret:// section in docs/secret-management.md** -- 80-line section added with subsections: URI format, storing secrets, referencing secrets in config, bootstrap constraint, resolution order, error handling (including struct field path in error messages).
4. **llms-full.txt** -- 4-line addition in the config section covering URI support, resolution timing, store command, and bootstrap constraint.
5. **AGENTS-VIBEWARDEN.md** -- Both the static copy (`docs/examples/AGENTS-VIBEWARDEN.md`) and the Go template (`internal/cli/templates/agents/agents-vibewarden.md.tmpl`) updated: old "env vars or secrets plugin" text replaced with `secret://` URI guidance, YAML example, and bootstrap constraint.
6. **ADR version consistency** -- README summary and ADR title/issue/status are aligned. No conflicts.

**Cross-file consistency verified**: URI format (`secret://path/key`) and bootstrap constraint language are consistent across all 5 documentation surfaces.

No remaining doc issues. Ready to merge from a documentation perspective.
